### ISSUE TITLE
values genereted with assign can only be used once

### ISSUE THREAD

**Issue by picaron** [fcsonline/drill#11]
it seems that the first request that use an assigned value consumes, I cannot reuse it in subsequent requests ...

**Comment by fcsonline** [comment]
I have checked this and I can reproduce it. This is my test plan with two kinds of _assign_:

```
---

threads: 2
base: 'http://localhost:9000'
iterations: 1

plan:
  - name: Fetch account
    request:
      url: /api/account
    assign: foo

  - name: Fetch manager by foo
    request:
      url: /api/users/{{ foo.manager_id }}

  - name: Fetch manager by foo (2)
    request:
      url: /api/users/{{ foo.manager_id }}

  - name: Assign values
    assign:
      key: bar
      value: "3"

  - name: Fetch manager by bar
    request:
      url: /api/users/{{ bar }}

  - name: Fetch manager by bar (2)
    request:
      url: /api/users/{{ bar }}
```

And this is my log:

```yml
Threads 1
Iterations 1
Base URL http://localhost:9000

Fetch account             http://localhost:9000/api/account 200 OK 2ms
Fetch manager by foo      http://localhost:9000/api/users/73 200 OK 1ms
Fetch manager by foo (2)  http://localhost:9000/api/users/73 200 OK 1ms
Assign values             bar=3
Fetch manager by bar      http://localhost:9000/api/users/3 200 OK 1ms
Fetch manager by bar (2)  http://localhost:9000/api/users/3 200 OK 1ms
```

As you can see I can reuse _assigns_.

**Comment by fcsonline** [comment]
I just spotted a bug regarding the priority of `Cookie` header. `drill` manages sessions automatically if you do some kind of login and remembers the `Set-Cookie` header. This was overwriting the specification for the second request in your test plan. Check this commit out:

https://github.com/fcsonline/drill/commit/f91cd503d22d93203e79c9f1b00a08b1685828dd

I released a new version: `0.3.8`

**Comment by picaron** [comment]
updated to version 0.3.8, your fix worked!

thanks for the super fast response and fix 👍
